access-common/src/main/java/edu/unc/lib/dl/ui/service/SolrQueryLayerService.java

```diff
@@ -338,152 +340,63 @@ public SearchResultResponse getFacetList(SearchRequest searchRequest) {
 
 			facetFieldUtil.addToSolrQuery(ancestorPath, solrQuery);
 		}
-
-		// If this item has no display order, get arbitrary items surrounding it.
-
-		Long pivotOrder = metadata.getDisplayOrder();
-
-		if (pivotOrder == null) {
-
-			LOG.debug("No display order, just querying for " + windowSize + " siblings");
-
-			StringBuilder query = new StringBuilder();
-
-			List<BriefObjectMetadataBean> list = null;
-
-			query.append("*:*");
-			query.append(accessRestrictionClause);
-			solrQuery.setQuery(query.toString());
-
-			solrQuery.setStart(0);
-			solrQuery.setRows(windowSize);
-
-			solrQuery.setSort(solrSettings.getFieldName(SearchFieldKeys.DISPLAY_ORDER.name()), SolrQuery.ORDER.desc);
-
-			try {
-				QueryResponse queryResponse = this.executeQuery(solrQuery);
-				list = queryResponse.getBeans(BriefObjectMetadataBean.class);
-			} catch (SolrServerException e) {
-				LOG.error("Error retrieving Neighboring items: " + e);
-				return null;
-			}
-
-			return list;
-
-			// Otherwise, query for items surrounding this item.
-
-		} else {
-
-			LOG.debug("Display order is " + pivotOrder);
-
-			// Find the right and left lists
-
-			StringBuilder query;
-
-			List<BriefObjectMetadataBean> leftList = null;
-			List<BriefObjectMetadataBean> rightList = null;
-
-			solrQuery.setStart(0);
-			solrQuery.setRows(windowSize - 1);
-
-			// Right list
-
-			query = new StringBuilder();
-
-			query.append(solrSettings.getFieldName(SearchFieldKeys.DISPLAY_ORDER.name())).append(":[")
-					.append(pivotOrder + 1).append(" TO *]");
-			query.append(accessRestrictionClause);
-			solrQuery.setQuery(query.toString());
-
-			solrQuery.setSort(solrSettings.getFieldName(SearchFieldKeys.DISPLAY_ORDER.name()), SolrQuery.ORDER.asc);
-
-			try {
-				QueryResponse queryResponse = this.executeQuery(solrQuery);
-				rightList = queryResponse.getBeans(BriefObjectMetadataBean.class);
-			} catch (SolrServerException e) {
-				LOG.error("Error retrieving Neighboring items: " + e);
-				return null;
-			}
-
-			LOG.debug("Got " + rightList.size() + " items for right list");
-
-			// Left list
-
-			// (Note that display order stuff is reversed.)
-
-			query = new StringBuilder();
-
-			query.append(solrSettings.getFieldName(SearchFieldKeys.DISPLAY_ORDER.name())).append(":[* TO ")
-					.append(pivotOrder - 1).append("]");
-			query.append(accessRestrictionClause);
-			solrQuery.setQuery(query.toString());
-
-			solrQuery.setSort(solrSettings.getFieldName(SearchFieldKeys.DISPLAY_ORDER.name()), SolrQuery.ORDER.desc);
-
+		
+		solrQuery.setQuery("*:*" + accessRestrictionClause);
+		
+		addSort(solrQuery, "default", true);
+		solrQuery.setRows(NEIGHBOR_SEEK_PAGE_SIZE);
+		solrQuery.setFields("id");
+		
+		long total = -1;
```

Comment:
This isn't super complicated, but a short note about what's going on here would be helpful. eg, "Query for ids in this container in groups of NEIGHBOR_SEEK_PAGE_SIZE until we find the offset of the object. Then query for windowSize objects with the start adjusted so that the object is in the middle."